app/models/work.rb

```diff
@@ -629,5 +638,14 @@ def embargo_date_as_json
         embargo_date_iso8601.gsub(/\+.+$/, "Z")
       end
     end
+
+    def log_performance(start, message)
+      elapsed = Time.zone.now - start
+      if elapsed > 20
```

Comment:
We can tweak the threshold, but I thought it would be good to start logging as warning slow requests